pandas/core/groupby.py

```diff
@@ -1923,8 +1923,16 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
 
             # a passed Categorical
             elif isinstance(self.grouper, Categorical):
+
+                # must have an ordered categorical
+                if self.sort:
+                    if not self.grouper.ordered:
+                        raise ValueError("cannot sort by an unordered Categorical in the grouper\n"
```

Comment:
This will make this awkward:

```
df = DataFrame({ 'A' : list('aabc'), 'C' : np.arange(4) })
df["B"]= df["A"].astype('category')
df.groupby('A').sum() # works
df.groupby('B').sum() # raises
```